core/src/main/java/hudson/triggers/SCMTrigger.java

```diff
@@ -351,21 +351,6 @@ public boolean isPollingThreadCountOptionVisible() {
             queue.setExecutors(Executors.newFixedThreadPool(maximumThreads, threadFactory()));
         }
 
-        @Override
-        public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
-            String t = json.optString("pollingThreadCount",null);
-            if (doCheckPollingThreadCount(t).kind != FormValidation.Kind.OK) {
-                setPollingThreadCount(THREADS_DEFAULT);
```

Comment:
Semantics change: This resets to default when invalid from user submission, the setter goes to the nearest boundary.

Also, don't we need a `@DataboundSetter` for this to work?

Reply:
Semantic here indeed was heterogeneous. descriptor didn't used the same strategy when configured by webUI and direct setter invocation. 
I'd personally would prefer it reject improper values vs silently prefer a sane one without caller to know his proposed value was rejected, but this goes beyond this PR.

Reply:
bindJSON(Object, json) do rely on javabean setters, no need for `@DataboundSetter` annotation - while this is required for Describable relying on bindJSON(class, json). Don't ask me about the logic here :)